### ISSUE TITLE
Extraction of VCS revisions in sub-directories

### ISSUE THREAD

**Issue by larskuhtz** [alephcloud/hs-configuration-tools#43]
The generated `PkgInfo` modules check for the revision control system with the following code.

```haskell
getVCS :: IO (Maybe RepoType)
getVCS =
    doesDirectoryExist ".hg" >>= \x0 -> if x0
    then return (Just Mercurial)
    else doesDirectoryExist ".git" >>= \x1 -> return $ if x1
        then Just Git
        else Nothing
```

This code only works correctly if called in the root directory of the repository. It's a common practice to develop more than one cabal package in a single repository by placing each package in a sub-directory. For that we must support detection of the VCS in sub-directories.

```haskell
getVCS :: IO (Maybe RepoType)
getVCS =
    doesDirectoryExist ".hg" >>= \x0 -> if x0
    then return (Just Mercurial)
    else doesDirectoryExist ".git" >>= \x1 -> return $ if x1
        then Just Git
        else Nothing {- TODO if pwd is not call getVCS on ".." -}
```

**Comment by larskuhtz** [comment]
Thanks @gregwebs for reporting this issue!

**Comment by larskuhtz** [comment]
Solved by #44
